src/mpi/pt2pt/recv.c

```diff
@@ -145,6 +145,19 @@ int MPI_Recv(void *buf, int count, MPI_Datatype datatype, int source, int tag,
         goto fn_exit;
     }
 
+    if (unlikely(MPIR_CVAR_ENABLE_FT &&
```

Comment:
These checks will eventually be needed in progress for ULFM, no? @yfguo @wesbland can you confirm?

Reply:
Yes, it breaks the semantics of ULFM. If a communication fails before completing due to process failure, we need to be able to handle it.

Reply:
This isn't process failure right ? This is comm anysource stuff, can be detected before progress ?

Reply:
Well, what ends up happening is if a process failure is detected, it will manifest in this check as anysource getting disabled: `!MPID_Comm_AS_enabled(request_ptr->comm)`. We still need to handle the case where we are waiting on an any source request and detect that a process fails, so we know to cancel that operation.

Reply:
I see, interesting, then I guess the only way to handle this is to move it all together into MPID_Wait ? sounds reasonable ?

Reply:
Please take a look again.